packages/foam-workspace-manager/src/core.ts

```diff
@@ -0,0 +1,89 @@
+import { Graph, Edge } from 'graphlib'
+
+
+type ID = string
+
+export interface FoamLink {
+  from: ID
+  to: ID
+  text: string
+}
+
+export interface OutLink {
+  to: ID
+  text: string
+}
+
+
+export class Bubble {
+  public id: ID
+  public title: string
+  public source: string
+  public path: string
+  public links: OutLink[]
+
+  constructor(id: ID, title: string, links: OutLink[], path: string, source: string) {
+    this.id = id
+    this.title = title
+    this.source = source
+    this.path = path
+    this.links = links
+  }
+}
+
+
+export class Foam {
+  private graph: Graph
+
+  constructor() {
+    this.graph = new Graph()
+  }
+
+  public setBubble(bubble: Bubble) {
+    if (this.graph.hasNode(bubble.id)) {
+      (this.graph.outEdges(bubble.id) || []).forEach(edge => {
```

Comment:
basically this cleans the old edges before adding the new ones, otherwise the old links would still exist - I might actually move that out and have a `removeBubble` (which we would need when removing a file anyways)